### ISSUE TITLE
JAM's mam4-jax adapter depends on submodule paths outside upstream's semver contract

### ISSUE THREAD

**Issue by duncanwp** [climate-analytics-lab/jax-gcm#737]
**Not urgent, and not a complaint about the upstream release.** mam4-jax 0.4.0 was cut quickly at our request so that #726 could stop pinning a feature branch; a public API that does not yet cover everything a host model reaches for is the expected consequence of that, not an oversight. The exact pin below is a perfectly good mitigation and can stay indefinitely. This issue exists so the coupling gap is tracked somewhere that comes back around, rather than living in a `pyproject.toml` comment.

The `Mam4JaxMicrophysics` adapter (`jcm/physics/aerosol/jam/microphysics/mam4_jax.py`) imports most of what it needs by dotted submodule path:

```python
from mam4_jax.core import data
from mam4_jax.coupling import amicphys as _amicphys
from mam4_jax.coupling.amicphys import amicphys
from mam4_jax.physics.calcsize import calcsize
from mam4_jax.physics.wateruptake import wateruptake
```

Upstream states explicitly (`mam4_jax/__init__.py`) that only the **top-level** namespace is covered by semantic versioning, and that submodule paths "are internal and DO move" — v0.3.0 renamed `mam4_jax.processes.*` out of existence with no deprecation period, which is exactly the breakage this adapter took on PR #726.

Of the five imports above only `AmicphysParams` / `configure_gas_netprod` / `configure_pcarbon_aging` are re-exported at the top level today. The adapter cannot move to the public API because the pieces it drives are not there:

- `amicphys()` itself — jcm calls it directly rather than `run_step`, because jcm needs to `vmap` `calcsize → wateruptake → amicphys` over its own flattened cell axis.
- `calcsize` / `wateruptake` — same reason.
- `configure_condensation` — the backend/substep selector, not exported.
- `core.data` — `NUMPTR_AMODE` / `LMASSPTR_AMODE` and the cloud-borne mirrors, which the tracer packing is built from.

**Consequence.** A minor upstream bump can break jcm at import, and (worse) an index-table reshuffle in `core.data` could break it silently. The current mitigation is the exact `mam4-jax==0.4.0` pin in `pyproject.toml` plus the two `hasattr` capability guards in `__init__`, which is defensive but does not let us track upstream.

**What would fix it.** A low-priority upstream request to re-export the per-process entry points and the index tables from the top-level namespace (an `amicphys` / `calcsize` / `wateruptake` / `configure_condensation` export plus a documented accessor for the MAM4 index bookkeeping), then move the adapter onto them and relax the pin to a compatible-release specifier (`~=0.x`). Good candidates to fold into whatever the next mam4-jax release is for its own reasons — there is no need to cut one for this. Until then the pin stays exact and every bump is a deliberate change with the JAM adapter tests run against it.

Found while re-pinning #726 onto the mam4-jax v0.4.0 release; cross-linked from the `mam4` extra's comment in `pyproject.toml`.
